### ISSUE TITLE
Ask for permission without calling 'authenticate'

### ISSUE THREAD

**Issue by yucehasan** [aparajita/capacitor-biometric-auth#3]
Hi, I believe there is no way to do this currently, so this is a feature request. We have a settings page in our app where we ask users whether they want to use biometrics or not. If they enable, we call `authenticate` on app login and FaceId permission is asked here. It would be good to have a way to ask for FaceId permission in settings rather than the login page when they actually attempt to use it for the first time. I believe this would improve the UX. Thanks in advance 🙏

**Comment by aparajita** [comment]
The plugin has no control over the UI of your app. If the user has not enabled biometric login in the settings, you shouldn't be calling `authenticate` during login.

**Comment by yucehasan** [comment]
Let me rephrase: I do not want the plugin to control my app. I just want to be able to use the plugin to ask for system level biometrics permission. For example, on iOS, when I first call `authenticate`, a native popup shows up and tells: "MyApp would like to use FaceID, do you want to allow?". I want to be able to do this without calling `authenticate`. Because it becomes weird if a user allows biometrics in my app settings (not in system settings) and then they go to lock screen and rejects it. Or, let's say they allowed it both in system and app settings but user decided to go to system settings and remove the permission. I think it would be good to add `hasPermission` and `requestPermission` methods in that case

**Comment by aparajita** [comment]
> when I first call authenticate, a native popup shows up and tells: "MyApp would like to use FaceID, do you want to allow?". I want to be able to do this without calling authenticate.

Can't be done. That is a system-level prompt that only appears the first time the user tries to authenticate with FaceID.

> Because it becomes weird if a user allows biometrics in my app settings (not in system settings) and then they go to lock screen and rejects it.

That's why you should be calling `checkBiometry` before even allowing the user the option to enable biometry in your app. If it isn't enabled or the user isn't enrolled at the system level, you shouldn't be offering the option in the app.

> let's say they allowed it both in system and app settings but user decided to go to system settings and remove the permission. I think it would be good to add hasPermission

That's why you need to call `checkBiometry` (which does exactly what your proposed `hasPermission` would do) on app resume. Please see the demo app for examples, I handled this there.

**Comment by yucehasan** [comment]
We are getting inconsistent result with`checkBiometry`. For some devices, it returns false even though device has the hardware and is enrolled. We are running on Capacitor 3. We are considering to update to Capacitor 4 if it would help. What do you suggest?

**Comment by aparajita** [comment]
Please open a new issue ("Inconsistent checkBiometry results") with more details about the exact model and OS of the devices, as well as what kinds of biometry the devices support.

**Comment by aparajita** [comment]
@yucehasan I can't help you if you don't give me more info.

**Comment by yucehasan** [comment]
@aparajita I am revisiting my code to make sure problem is not on my side. So far, it looks like the plugin is working fine. I will create new issue if I find anything. Thanks for your attention.

**Comment by aparajita** [comment]
The plugin is just reporting what the OS says, so it can't lie. The only possible issue I can think of is Android devices that have more than one biometry option, in which case I only report what the system considers the primary biometry.
